vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -105,15 +105,19 @@ impl PendingCommands {
         });
     }
 
-    fn remove(&mut self, cid: u16) -> Rpc<(), spec::Completion> {
+    fn remove(&mut self, cid: u16, queue_id: u16) -> Rpc<(), spec::Completion> {
```

Comment:
Let's try making this change without altering the function signature. I think that means changing the PendingCommands structure to contain a qid. Do not change any mesh save state.